django/db/backends/mysql/schema.py

```diff
@@ -48,7 +48,11 @@ def _is_limited_data_type(self, field):
         return db_type is not None and db_type.lower() in self.connection._limited_data_types
 
     def skip_default(self, field):
-        return self._is_limited_data_type(field)
+        # MySQL and MariaDB < 10.2.1 don't support defaults for BLOB or TEXT
+        # https://mariadb.com/kb/en/library/blob-and-text-data-types/
+        if not self.connection.mysql_is_mariadb or self.connection.mysql_version < (10, 2, 1):
```

Comment:
Maybe we could add a cached property with this check in `schema.py`, e.g.
```python
@cached_property
def _support_default_on_limited_data_type(self):
    return self.connection.mysql_is_mariadb and self.connection.mysql_version > (10, 2)
```

Reply:
I don't think `cached_property` is worth it, it's a really fast few binary comparisons - the underlying properties are cached.

Reply:
So maybe without cache, with a separate property we will be able to fix in only one place after [30712](https://code.djangoproject.com/ticket/30712).

Reply:
Done